enhancements/dns/forward-dns-over-tls.md

```diff
@@ -0,0 +1,317 @@
+can set the following:
+* `spec.servers[].forwardPlugin.transport`
+* `spec.servers[].forwardPlugin.caBundle`
+* `spec.servers[].forwardPlugin.servername`
+
+Example:
+```yaml
+spec:
+  servers:
+  - name: foo-server-with-tls
+    zones:
+      - foo.com
+    forwardPlugin:
+      transport: tls
+      tls:
+        caBundle: dns-tls-configmap
+        servername: dns.bar.com
+      upstreams:
+        - 1.1.1.1
+        - 2.2.2.2:5353
+```
+
+### API Extensions
+
+The `UpstreamResolvers` and `ForwardPlugin` APIs will be modified to allow 
+for TLS configuration.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Implementing this enhancement requires changes in the following repositories:
+
+openshift/api
+openshift/cluster-dns-operator
+
+### Risks and Mitigations
+
+**Risk 1:**
+* Moving away from CoreDNS could impact the API
+
+**Mitigations:**
+* Ensure the API is flexible enough to handle reasonable digression from 
+  what CoreDNS offers in the `forward` plugin.
+
+**Security:** Given that this enhancement is inherently security related, 
+security review of the approach will be preferred.
+
+**UX:** Given that this enhancement is only accessible via configuration 
+editing, no UX testing is required.
+
+## Design Details
+
+### Open Questions [optional]
+
+Q. Should the cluster administrator supplied certificates be stored in the openshift-dns or openshift-dns-operator namespaces?
+
+A. "The administrator should put the configmaps in the openshift-config namespace, and the operator should copy the configmaps from there to the openshift-dns namespace so the DNS pods can use them in volume mounts."
+
+### Test Plan
+
+A new e2e test will be added; TestDNSOverTLSForwarding. This can reuse 
+most of the existing logic from the TestDNSForwarding e2e test. The 
+implementation is intended to be relatively straightforward so no special 
+testing considerations are known at this time aside from the need for an 
+upstream resolver that has DNS-over-TLS enabled.
+
+### Graduation Criteria
+
+N/A
+
+#### Dev Preview -> Tech Preview
+
+N/A.  This feature will go directly to GA.
+
+#### Tech Preview -> GA
+
+N/A.  This feature will go directly to GA.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+On upgrade, the default configuration and existing forward configuration 
+will work without issue.
+
+On downgrade, this functionality will revert to sending plaintext 
+DNS queries since the downgraded operator will have the older CRD 
+and logic that does not include these new API fields. Cluster 
+administrators should remove configuration specific to this enhancement 
+after downgrade.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
+
+* **Misconfiguration of the TLS certs**
+  * Cluster administrators can revert the configuration.
+  * TLS certs will be validated prior to admission using existing code or
+    libraries.
```

Comment:
The phrase "validated prior to admission" suggests to me that you intend to implement an API admission check, but the user creates the configmap using an arbitrary name and may create the configmap before adding a reference to it from the cluster dns config.  What kind of admission check do you mean, and how do validation errors get reported?

Reply:
I've removed this as it's not necessary. CoreDNS logs cert related errors when trying to establish a connection.